decodeSignal.m

```diff
@@ -1,4 +1,23 @@
-function [decoded, actualBER] = decodeSignal(signal)
-  actualBER = 0.03;
-  decoded = [0 1 0 1]; 
-end
\ No newline at end of file
+function [decoded,actualBER] = decodeSignal(signal)
+    signalLength = length(signal);
+    decoded = zeros(signalLength/3,1);
+    j=1; 
+    errors =0 ;
+    for i=1:3:signalLength
+        sum = signal(i)+signal(i+1)+signal(i+2);
+        if sum <2
+            if sum ==1
+                errors = errors+1;
+            end
+            decoded(j) = 0;
+        else
+            if sum ==2
+                errors = errors +1;
+            end
+            decoded(j) = 1;
+        end
+        j=j+1;
+    end
+    actualBER = errors/length(decoded);
```

Comment:
@krzysiekissel actualBER liczymy raczej na dlugosci sygnalu przesyłanego (signalLength)

Reply:
Jeśli tak, to do obliczenia actualBER jest potrzebny pierwotny sygnał bez zakłóceń, bo jak odróżnić czy w ciągu 011 jest walnięte 0  czy dwie 11 ? W prawdzie teraz tez to jest tak zrobione ze przy ciągu 000 nie zlicza błędu a przeciez mogły być wszystkie 3 zera przekłamane.

Reply:
Też się zastanawiałem to już jest inny typ BERa. Można by wyznaczyć w sumie 3 typy BER. Ale myśląc dłużej to BER służy tylko do wyznaczania błędów w przesyłaniu. Przy tym sposobie kodowania jest możliwy błąd i chyba trzeba to akceptować. Reasumując actualBER liczymy na sygnale z zakłóceniami - może się on różnic z względu na małą liczbę bitów lub duży BER przesyłanych gdzie jest wieksze prawdopodobieństwo, że dwa bity obok siebie się zamienią

Reply:
A może zamiast liczenia BERa w funkcji decode, napiszę kolejna funkcje której argumentami będą sygnały przed zakłócenami i po zakłóceniach i tam policzę ten actualBER? Tak wydaje mi się rozdzielimy funkcję na takie które biorą udział w transmisji danych i na takie które mierzą jak dobra była transmisjia, takie bardziej pomiarowe. I tam bym mógł też wykresy porobić itp.

Reply:
W sumie actualBER jest w miarę ściśle związany z tą funkcją. Gdybyś chciał ją rozdzielić musiałbyś nieźle pokombinować z uwagi na matlaba lub podwoić kod. Pod wykresy osobne funkcje się zrobi, może funkcje można chować do folderów to by się pochowało /first-stage oraz /presentation a main w root folderze

Reply:
Oki, to pisze zeby zliczało, że jeżeli jest 011 lub 100 to dwa błędne bity a np 101 to jeden błędny bit tak ?

Reply:
Nie rozumiem czemu jaka jest różnica miedzy 011 a 101. Tutaj losowość jest na którą nie mamy pewności. Za kazdym razem gdy mamy tylko 3 bity możemy stwierdzić że jeden lub zero zostało przekłamanych. Gdy będziemy porównywać z sygnałem początkowym możemy coś więcej powiedzieć ale nie dostaliśmy wytycznych